src/libstore/build/derivation-goal.cc

```diff
@@ -1207,7 +1207,14 @@ void DerivationGoal::handleChildOutput(int fd, const string & data)
     if (hook && fd == hook->fromHook.readSide.get()) {
         for (auto c : data)
             if (c == '\n') {
-                handleJSONLogMessage(currentHookLine, worker.act, hook->activities, true);
+                auto s = handleJSONLogMessage(currentHookLine, worker.act, hook->activities, true);
+                if (s && !isWrittenToLog && logSink) {
+                    auto json = nlohmann::json::parse(std::string(currentHookLine, 5));
```

Comment:
Note: we're effectively parsing each message twice with this patch. Any suggestions how to restructure the code to avoid that in a reasonable way (not too experienced with the codebase admittedly)? Or is this neglectable?

Reply:
Good catch, yes I think we should fix this but I'll have to check not ob my phone to see how.